Automattic/newspack-plugin
# Stripe: create native WC Subscriptions (if a flag is set)

### All Submissions:






### Changes proposed in this Pull Request:



Changes the way Stripe donations are handled, so that it's possible that instead of a Stripe Subscription, a WC Subscriptions' Subscription is created for recurring payments. 

There is an underlying change in how Stripe API is interacted with. The way to charge a customer is either through [Sources](https://stripe.com/docs/api/sources), [Tokens](https://stripe.com/docs/api/tokens), or [Payment Methods](https://stripe.com/docs/api/payment_methods) – the latter [replaces the former two](https://stripe.com/docs/payments/payment-methods/transitioning). This integration has used Payment Methods, but since [WC Stripe Gateway uses a Source to process WC Subscriptions renewals](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/compat/trait-wc-stripe-subscriptions.php#L245-L247), the integration was updated to use Sources instead of Payment Methods.  

### How to test the changes in this Pull Request:

There are two ways to pay, and two modes of payment. When testing, ideally test all four scenarios (let's name them with letters for easier communication):

|         | Once | Recurring (e.g. monthly) 
|--------------|-----------|--|
| Card form | A | B  | 
| PaymentRequest (e.g. Apple Pay)      | C | D

1. Set up Stripe, ensure the site is reachable by Stripe webhooks
2. Ensure the WC suite is active, and "Stripe – Credit Card (Stripe)" set as the sole payment method in WC
1. Test the payment scenarios for regressions – the recurring payments should result in Stripe Subscriptions shadowed by sync'd WC Subscriptions (which can't be edited, etc.)
3. Set the `NEWSPACK_USE_WC_SUBSCRIPTIONS_WITH_STRIPE_PLATFORM` environment flag to `true`
4. Test the payment scenarios with this new configuration – recurring payments should result in WC Subscriptions and one-time charges in Stripe (_not_ Stripe Subscriptions)
5. Open the WC Subscriptions (scenarios B & D) in WP admin renew them manually (via the "Subscription actions" meta box)
7. Observe that the two subscriptions were renewed:
    1. Each got a new Stripe charge (successful)
    2. Each got a renewal order in WC
8. Disable the Stripe gateway in WC, observe that a recurring donation results in a Stripe Subscription

### Other information:
